cmake/checks/check_02_system_features.cmake

```diff
@@ -126,18 +126,21 @@ IF(CMAKE_SYSTEM_NAME MATCHES "Windows")
   #
   SET(DEAL_II_MSVC TRUE)
 
+
   #
   # Shared library handling:
   #
+  IF(NOT "${CMAKE_VERSION}" VERSION_LESS "3.4")
+    #
+    # CMake version 3.4 or newer supports automatic symbol exports under
+    # Windows. Enable this feature for all targets unconditionally:
+    #
+    SET(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS TRUE)
```

Comment:
Yay -- this is why I think `cmake` is such a cool project, because they implement things everyone seems to need!